nimbus/src/lib.rs

```diff
@@ -154,6 +167,25 @@ impl NimbusClient {
     }
 }
 
+pub trait SchemaRequirement {
+    fn schema_requirement(&self) -> VersionReq;
+}
```

Comment:
I'm not sure this needs to be a trait, as you don't use it as a trait bound anywhere. Would it work if you just `impl NimbusClient` rather than `impl SchemaRequirement for NimbusClient` below?